src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -154,11 +174,22 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
       {
         if (!node->isLeaf())
         {
-          next = node->left_child;
+          int const left_child_index = node->left_child;
+          if (!pruneLeftSubtree(queryIndex, left_child_index))
+          {
+            next = left_child_index;
+          }
+          else
+          {
+            Node const *left_child_node = _bvh.getNodePtr(left_child_index);
```

Comment:
This is really unfortunate, as we still pay for loading the left child. Don't know how to avoid it, though.